src/v/rpc/test/rpc_gen_cycling_test.cc

```diff
@@ -454,3 +454,64 @@ FIXTURE_TEST(corrupted_data_at_server, rpc_integration_fixture) {
         BOOST_REQUIRE_EQUAL(echo_resp_new.value().data.str, "testing...");
     }
 }
+
+FIXTURE_TEST(version_not_supported, rpc_integration_fixture) {
+    configure_server();
+    register_services();
+    start_server();
+
+    rpc::transport t(client_config());
+    t.connect(model::no_timeout).get();
+    auto client = echo::echo_client_protocol(t);
+
+    const auto check_unsupported = [&] {
+        auto f = t.send_typed_versioned<echo::echo_req, echo::echo_resp>(
+          echo::echo_req{.str = "testing..."},
+          960598415,
```

Comment:
nit: unrelated to this patch, and feel free to punt. I know/think this is codegened, but maybe it's worth still defining some named constexpr variables for method IDs. Maybe we can codegen some methods that give the IDs.

Reply:
yeh I wish the code generator created a enum or something for these. it should be straight forward to add

Reply:
This is starting to get out of scope for the PR. I added an issue here to track this improvement https://github.com/redpanda-data/redpanda/issues/4892